core/src/org/labkey/core/wiki/HtmlRenderer.java

```diff
@@ -225,118 +226,135 @@ public FormattedHtml format(String text)
+            case Substitute -> handleSubstitutions(text);
+        };
+    }
 
     // ${labkey.<type>(<any_stream of characters>)}
     // Pattern.DOTALL allows the parameter list to span multiple lines
-    private static final Pattern _substitutionPattern = Pattern.compile("\\$\\{labkey\\.(\\w+)\\((.*?)\\)}", Pattern.DOTALL);
+    private static final Pattern SUBSTITUTION_PATTERN = Pattern.compile("\\$\\{labkey\\.(\\w+)\\((.*?)\\)}", Pattern.DOTALL);
 
-    // <any word>='<any value>', allowing whitespace before, after, and in-between
-    private static final Pattern _paramPattern = Pattern.compile("\\s*(\\w+)\\s*=\\s*'(.*)'\\s*");
+    // Find all substitution templates embedded in HTML wiki text that have the form ${labkey.<type>(<any_stream of characters>)}.
+    private Matcher getSubstitutionMatcher(String text)
+    {
+        return SUBSTITUTION_PATTERN.matcher(text);
+    }
 
-    private FormattedHtml handleLabKeySubstitutions(String text)
+    private FormattedHtml removeSubstitutions(String text)
     {
-        if (text == null)
-            return new FormattedHtml(HtmlString.EMPTY_STRING);
+        text = getSubstitutionMatcher(text).replaceAll("");
+        return new FormattedHtml(HtmlString.unsafe(text));
+    }
 
+    // <any word>='<any value>', allowing whitespace before, after, and in-between
+    private static final Pattern PARAM_PATTERN = Pattern.compile("\\s*(\\w+)\\s*=\\s*'(.*)'\\s*");
+
+    private FormattedHtml handleSubstitutions(String text)
+    {
         boolean volatilePage = false;
         LinkedHashSet<ClientDependency> cds = new LinkedHashSet<>();
 
-        if (_allowSubstitutions)
-        {
-            // Find all substitution templates embedded in wiki text that have the form ${labkey.<type>(<any_stream of characters>)}.
-            Matcher webPartMatcher = _substitutionPattern.matcher(text);
+        Matcher substitutionMatcher = getSubstitutionMatcher(text);
 
-            // If we find none, return immediately
-            if (!webPartMatcher.find())
-                return new FormattedHtml(HtmlString.unsafe(text));
+        // If we find none, return immediately
+        if (!substitutionMatcher.find())
+            return new FormattedHtml(HtmlString.unsafe(text));
 
-            List<Definition> definitions = new ArrayList<>(10);
-            Map<Definition, List<String>> wikiErrors = new HashMap<>();
-            do
+        List<Definition> definitions = new ArrayList<>(10);
+        Map<Definition, List<String>> wikiErrors = new HashMap<>();
+        do
+        {
+            List<String> paramErrors = new ArrayList<>();
+            String substitutionType = substitutionMatcher.group(1);          // type
+            String params = substitutionMatcher.group(2).replace(",", "");
+            // Parse the parameters with the symbols in parseWith, they can be used in any order
+            // as long as they are the same symbol starts and completes a parameter value
```

Comment:
This doesn't read correctly. Is this the intended version?

```suggestion
            // as long as the same symbol starts and completes a parameter value
```

Reply:
Old comment, but sure